src/mongo-db/user.schema.ts

```diff
@@ -1,14 +1,21 @@
 /* eslint-disable prettier/prettier */
 import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
-import { HydratedDocument } from 'mongoose';
+import mongoose, { HydratedDocument } from 'mongoose';
 
 export type UserDocument = HydratedDocument<User>;
 
 @Schema()
 export class User {
   @Prop()
-  name: string;
+  first_name: string;
+
+  @Prop()
+  last_name: string;
+
   @Prop()
   age: number;
+
+  @Prop({ type: mongoose.Schema.Types.ObjectId, ref: 'User' })
```

Comment:
This is the user, should not refer to itself, the user can point avatar as a content reference

avatar: Content<Ref>